lib_xua/host/xmosdfu/src/xmosdfu.cpp

```diff
@@ -82,104 +65,108 @@ enum dfu_state {
 };
 
 static libusb_device_handle *devh = NULL;
+static int device_bInterfaceProtocol;
+static int match_vendor = -1;
+static int match_product = -1;
+static int match_vendor_dfu = -1;
+static int match_product_dfu = -1;
 
-static int find_xmos_device(unsigned int id, unsigned int pid, unsigned int list)
-{
-    libusb_device *dev;
-    libusb_device **devs;
-    int i = 0;
-    unsigned int found = 0;
 
-    size_t count = libusb_get_device_list(NULL, &devs);
-    if ((int)count < 0)
-    {
-        printf("ERROR: get_device_list returned %d\n", (int)count);
-        exit(1);
+static int probe_configuration(libusb_device *dev, struct libusb_device_descriptor *desc, unsigned int list)
+{
+    struct libusb_config_descriptor *config_desc = NULL;
+    int ret = libusb_get_active_config_descriptor(dev, &config_desc);
```

Comment:
You need to use `libusb_free_config_descriptor` to release the memory allocated by this function.

Reply:
done